lib/core/src/Cardano/Wallet/DB/Model.hs

```diff
@@ -291,29 +291,81 @@ mReadWalletMeta wid db@(Database wallets _) =
     (Right (mkMetadata <$> Map.lookup wid wallets), db)
   where
     mkMetadata :: WalletDatabase s xprv -> WalletMetadata
-    mkMetadata WalletDatabase{certificates,metadata} =
-        case Map.lookupMax certificates of
-            Nothing ->
-                metadata { delegation = NotDelegating }
-            Just (_, Nothing) ->
-                metadata { delegation = NotDelegating }
-            Just (_, Just pool) ->
-                metadata { delegation = Delegating pool }
-
-mReadWalletDelegations
-    :: Ord wid
-    => wid
-    -> ModelOp wid s xprv [DelegationDiscovered]
-mReadWalletDelegations wid db@(Database wallets _) = (Right res, db)
-  where
-    res = maybe mempty mkDelegations $ Map.lookup wid wallets
-    mkDelegations :: WalletDatabase s xprv -> [DelegationDiscovered]
-    mkDelegations WalletDatabase{certificates} =
-        map toDelegationDiscovered $
-        L.sortOn (Down . fst) $
-        Map.toList certificates
-    toDelegationDiscovered (slotId, poolIdM) =
-        DelegationDiscovered slotId poolIdM
+    mkMetadata WalletDatabase{checkpoints,certificates,metadata} =
```

Comment:
this stuff needs further refactoring and enhancement. Now it passes every time, but it is because the series of events : two or more delegation certificates inserted within two adjacent epochs (X, X+1) AND checkpoint at (X+1) is so very much improbable. We should introduce this kind of event to state machine to properly test this cases. Then, `mReadWalletDelegations` should fail as it produces different result that Sqlite implementation